diff -urpN klippy-stock/extras/shaper_calibrate.py klippy/extras/shaper_calibrate.py
--- klippy-stock/extras/shaper_calibrate.py	2023-10-17 20:01:35.123889133 -0400
+++ klippy/extras/shaper_calibrate.py	2023-10-17 20:02:43.803622708 -0400
@@ -4,7 +4,7 @@
 #
 # This file may be distributed under the terms of the GNU GPLv3 license.
 import collections, importlib, logging, math, multiprocessing, traceback
-shaper_defs = importlib.import_module('.shaper_defs', 'extras')
+shaper_defs = importlib.import_module('.shaper_defs', 'klippy.extras')
 
 MIN_FREQ = 5.
 MAX_FREQ = 200.
